lib/src/over_react_redux/devtools/middleware.dart

```diff
@@ -0,0 +1,143 @@
+  external send([action, state]);
+  external init([initalState]);
+}
+
+@JS('__REDUX_DEVTOOLS_EXTENSION__.connect')
+external ReduxDevToolsExtensionConnection reduxExtConnect([dynamic options]);
+
+class OverReactReduxDevToolsMiddleware extends MiddlewareClass {
+  bool devToolsExtensionFound = false;
+  Store _store;
+  ReduxDevToolsExtensionConnection devToolsExt;
+  OverReactReduxDevToolsMiddleware() {
+    try {
+      devToolsExt = reduxExtConnect();
+      devToolsExtensionFound = true;
+    } catch(e) {
+      print('Unable to connect to the redux dev tools browser extension. \nPlease install it... \nChrome: https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en \nFirefox: https://addons.mozilla.org/en-US/firefox/addon/reduxdevtools/');
+    }
+
+  }
+
+  set store(Store v) {
+    _store = v;
+    devToolsExt.init(_encodeForTransit(store.state));
+    devToolsExt.subscribe(handleEventFromRemote);
+  }
+
+  get store => _store;
+
+  dynamic _encodeForTransit(dynamic content) {
+    try {
+    return jsify(jsonDecode(jsonEncode(content)));
+    } catch (e) {
+      print('You must implement a `toJson` method in your state and actions in order to view state changes in the redux dev tools.');
+    }
+  }
+
+  dynamic _encodeActionForTransit(dynamic action) {
+    try {
+      return _encodeForTransit({'type': _getActionType(action), 'payload': action});
+    } on Error {
+      return _encodeForTransit({'type': _getActionType(action)});
+    }
+  }
+
+  String _getActionType(dynamic action) {
+    if (action.toString().contains('Instance of')) {
+      return action.runtimeType.toString();
+    }
+    return action.toString();
+  }
+
+  _relay(String type, [Object state, dynamic action, String nextActionId]) {
+    var message = JsBackedMap();
+    message['type'] = type;
+
+    if (state != null) {
+      try {
+        message['payload'] = _encodeForTransit(state);
+      } catch (error) {
+        message['payload'] = 'Could not encode state. Ensure state is json encodable';
+      }
+    }
+    if (type == 'ACTION') {
+      message['action'] = _encodeActionForTransit(action);
+      message['nextActionId'] = nextActionId;
+    } else if (action != null) {
+      message['action'] = action as String;
+    }
+
+    if (message.containsKey('action')) {
+      devToolsExt.send(message['action'], message['payload']);
+    } else {
+      devToolsExt.send(message.jsObject);
+    }
+  }
+
+  void handleEventFromRemote(JsMap jsData) {
+    JsBackedMap data = JsBackedMap.fromJs(jsData);
+    switch (data['type'] as String) {
+      case 'DISPATCH':
+        _handleDispatch(JsBackedMap.fromJs(data['payload']));
+        break;
+      case 'ACTION':
+        _handleRemoteAction(data['action'] as String);
+        break;
+      default:
+        print('Unknown type:' + data['type'].toString());
+    }
+  }
+
+  void _handleDispatch(dynamic action) {
+    if (this._store == null) {
```

Comment:
Yeah... it is a bad thing lol but it can happen... previously users has to do`whateverMiddleware.store = Store(` but now if you look down in call i actually get passed the store and so on the very first call to this middleware I set the store for use to the instance if its null so users no longer are required to set it at all :)